### PR TITLE
APICLI-1284: Add Trigger Python Client Generation Workflow

### PR DESCRIPTION
**Pull request by danaelhe**
This workflow will trigger the workflow in [digitalocean/digitalocean-client-python](https://github.com/digitalocean/digitalocean-client-python/pull/5) to re-generate the client when a change is done the spec.

### CODE REVIEW

File: .github/workflows/trigger-python-client-gen.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+jobs:
+  build:
+    name: Trigger digitalocean-client-python Workflow
+    runs-on: ubuntu-latest
+    steps:
+      - uses: convictional/trigger-workflow-and-wait@v1.3.0
+        with:
+          owner: digitalocean
+          repo: digitalocean-client-python
+          github_token: ${{ secrets.OPENAPI_SECRET }}
```
Comment (danaelhe) [inline .github/workflows/trigger-python-client-gen.yml:15]: secret was created in this repo. token is stored in lastpass.

File: .github/workflows/trigger-python-client-gen.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+name: Trigger Python Client Generation
+on: 
+  push:
+    paths:
```
Comment (andrewsomething) [inline .github/workflows/trigger-python-client-gen.yml:4]:
```suggestion
on: 
  push:
    branches:
      - main
    paths:
```

Comment (andrewsomething) [reply .github/workflows/trigger-python-client-gen.yml:4]: Think we only want this to trigger for pushes to main (or the `operationID` feature branch for now).

File: .github/workflows/trigger-python-client-gen.yml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+      - 'specification/**'
+jobs:
+  build:
+    name: Trigger digitalocean-client-python Workflow
+    runs-on: ubuntu-latest
+    steps:
+      - uses: convictional/trigger-workflow-and-wait@v1.3.0
```
Comment (andrewsomething) [inline .github/workflows/trigger-python-client-gen.yml:11]: Looks like this blocks until the workflow completes in the other repo. Wonder if we need all its features? Always a bit wary of third-party actions having access to secrets. Surprised that GitHub doesn't offer something official for this. We could just hit the API endpoint.

I think the GitHub CLI is available on the runners. Using that, it should be something like:

    gh workflow run --repo digitalocean/digitalocean-client-python python-client-gen.yml --ref main

https://docs.github.com/en/rest/actions/workflows#create-a-workflow-dispatch-event

Comment (danaelhe) [reply .github/workflows/trigger-python-client-gen.yml]: that makes sense! I was able to replace it out seamlessly. thanks for the suggestion.

**Review by andrewsomething — APPROVED** [review]
:+1:  LGTM!

**Review by andrewsomething — APPROVED** [review]

**Review by scotchneat — APPROVED** [review]
👍🏽

**Pull request merged**